content/addons.md

```diff
@@ -68,27 +68,35 @@ ember addon <addon-name> [options]
 
 The result is the creation of a directory called `<addon-name>`, which has many files and looks a bit like an Ember app. We won't need to use all the files to make a useful addon. By convention, _most_ Ember addons start with `ember` in the name, like `ember-basic-dropdown`. This will help other developers find our addon.
 
-<!-- Should we cover in-repo addons at all??? -->
-<!-- 
-If the addon is just meant to be used in a single project, an "in-repo" addon could be created instead. The benefit is that it is lightweight, but there are some major limitations; an in-repo addon can't be shared between apps, versioned independently, or published to npm. From within an existing Ember app, use:
+### Addon file structure
 
-```bash
-ember generate in-repo-addon <addon-name> [options]
-```
+In some ways, an addon is like a mini Ember app. It has a very similar file structure, uses a lot of the same API methods, and can do most things that components are able to do. 
 
-This generates a folder called `lib/<addon-name>` that contains its own `package.json` and an `index.js` file. 
--->
+Let's take a look a some of the most important files and folders in an addon, and how they are different from what we would find in an app. 
 
-### Addon file structure
+#### `addon/` 
 
-In some ways, an addon is like a mini Ember app. It has a very similar file structure, uses a lot of the same API methods, and can do most things that components are able to do. 
+If an addon only offers one UI template, the addon files usually go in the `addon/` directory. When the addon is used in a template, it is called by the "real name" of the addon, like `{{my-addon-name someValue=true}}`. This directory can hold many of the same subdirectories and files that an Ember app would, like `/components/` and `/templates/`.
+
+#### `app/` 
+
+If an addon will export multiple templates, the addon's files are referenced from the `app/` directory. When the addon templates are used, they are called by the template name, not the addon name. For example, the addon could provide `{{super-hero-img}}` and `{{fantastic-form}}`. Many addon developers prefix these kinds of templates with a nickname in order to avoid namespace clashes. For example, an addon template named `{{table-row}}` might cause problems if the app using it has a component by the same name, so `{{bs-table-row}}` is safer.
+
+#### `tests/dummy/`
+This directory contains a full Ember app for addon testing purposes. During tests, we can check to make sure that the addon works or looks as expected when it is used in an app. Many addon developers use the dummy app to hold their documentation site's content as well.
+
+#### `package.json` 
+
+If we want other people to be able to use our addon, at minimum, we need to specify a license, a version, the repository url, and description.
```

Comment:
Repository URL isn't required but `name` is. It's also useful to add `keywords: ["ember-addon"]` to make it easier to discover on emberobserver, emberaddons etc.

Reply:
Generating an addon will add those keywords. Repository URL should be strongly encouraged (emberobserver cannot review without it).